### PR TITLE
[libc][math][c23] Implement fmaxf16 and fminf16 function

### PR DESCRIPTION
**Pull request by HendrikHuebner**
Implements fmaxf16 and fminf16, which are two missing functions listed here: #93566

### CODE REVIEW

**Comment by overmighty** [comment]
Thanks, but I'm not sure if I'm allowed to have other people work on this issue as it is part of a Google Summer of Code project: https://summerofcode.withgoogle.com/programs/2024/projects/1vPVUj93.

**Comment by HendrikHuebner** [comment]
Uhm... well it would certainly be good if that was stated in the issue somewhere. Maybe put [gsoc] in the title?

**Comment by overmighty** [comment]
From what I've seen, you're expected to ask to be assigned to an issue before opening a PR for it. You should see that I am listed in the "Assignees" list.

I'll ask GSoC support about this.

**Comment by lntue** [comment]
> Uhm... well it would certainly be good if that was stated in the issue somewhere. Maybe put [gsoc] in the title?
> 
> Could you check if this can be merged anyways?

Sorry for the confusion and it's totally my fault to not making the tracking issue cleared that it's for this year GSoC project.

@overmighty : can you update https://github.com/llvm/llvm-project/issues/93566 title and description to indicate that it is a tracking issue for this year GSoC project?

@HendrikHuebner : thanks for the patch, I'll review and merge the PR.  And let me know if you are interested in working on math functions for libc, so that we can coordinate to avoid duplicated work.  Direct messaging me on discord or on libc channel would work best for me.

File: libc/test/src/math/smoke/fmaxf16_test.cpp [hunk 1]
```diff
@@ -0,0 +1,13 @@
+//===-- Unittests for fmaxf128 --------------------------------------------===//
```
Comment (lntue) [inline libc/test/src/math/smoke/fmaxf16_test.cpp:1]: nit: fmaxf16

**Comment by overmighty** [comment]
> @overmighty : can you update #93566 title and description to indicate that it is a tracking issue for this year GSoC project?

Done.

File: libc/test/src/math/smoke/fminf16_test.cpp [hunk 2]
```diff
@@ -0,0 +1,13 @@
+//===-- Unittests for fminf -----------------------------------------------===//
```
Comment (lntue) [inline libc/test/src/math/smoke/fminf16_test.cpp:1]: nit: fminf16

File: libc/src/math/generic/fmaxf16.cpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
+//===-- Unittest of fmaxf16 function --------------------------------------===//
```
Comment (lntue) [inline libc/src/math/generic/fmaxf16.cpp:1]: nit: Implementation

**Review by lntue — COMMENTED** [review]
Overall, it looks good to me.  Can you also update the status page at `docs/math/index.rst`?

**Comment by HendrikHuebner** [comment]
@Intue sure, I'll update the comments and the status page later. If there are any non-gsoc math functions that need implementation let me know. I can do a few others.

**Comment by lntue** [comment]
> @lntue sure, I'll update the comments and the status page later. If there are any non-gsoc math functions that need implementation let me know. I can do a few others.

@HendrikHuebner can you comment on https://github.com/llvm/llvm-project/issues/94312 so that I can assign it to you?

**Comment by HendrikHuebner** [comment]
@Intue I added the suggested changes

**Comment by lntue** [comment]
> @lntue I added the suggested changes

Can you sync / rebase the PR to resolve conflicts?

**Comment by HendrikHuebner** [comment]
done

**Review by lntue — APPROVED** [review]

**Pull request merged**